articles/implementing-rate-limiting-in-flask/index.md

```diff
@@ -0,0 +1,234 @@
+
+![Flask API response](/engineering-education/implementing-rate-limiting-in-flask/oqahhiaqcfeigq1urdwe.png)
+
+### Implementing Rate Limiting in Flask
+
+After installing the Flask Framework and saving the code responsible for setting up the endpoints as described in the steps above, the next step here is to install the [Flask-Limiter](https://flask-limiter.readthedocs.io/en/stable/) library. This is a Flask extension that helps to easily implement rate limiting rules in a Flask application.
+
+In the terminal, type:
+
+```Bash
+pip install Flask-Limiter
+```
+
+Now, you need to update your `app.py` file and integrate the `Flask-Limiter` library to define rate limiting rules for certain endpoints in your API.
+
+#### Importing the Flask-Limiter Library
+
+```Python
+from flask_limiter import Limiter
+from flask_limiter.util import get_remote_address
+```
+
+#### Setting Up Flask-Limiter
+
+```Python
+limiter = Limiter(app, key_func=get_remote_address)
+```
+
+#### Applying Rate Limiting Rules
+
+```Python
+@app.route("/")
+@limiter.limit("10/minute") # maximum of 10 requests per minute
+def index():
+  return "Welcome to my Flask API"
+```
+
+In the code above, you defined the rate limiting rule for the endpoint by setting it to accept `10` requests per `60` seconds from every user of the API. If the limit is exceeded, the user receives the response - `status code 429 (Too Many Requests)` and an error page similar to the image below.
+
+![Flask-Limiter limit exceeded error](/engineering-education/implementing-rate-limiting-in-flask/osw9-4n2ilprrg2m_i9g.png)
+
+Flask-Limiter provides a set of string notations for defining rate limit rules in its [documentation](https://flask-limiter.readthedocs.io/en/stable/%23rate-limit-string-notation) with the given format:
+
+```
+[count] [per|/] [n (optional)] [second|minute|hour|day|month|year]
+```
+
+You can also combine multiple rate limits by separating them with a delimiter of your choice. Examples include:
+
+- 10 per second
+- 60/hour
+- 5/second;60 per minute;2500 per hour
+- 50/day, 250/7days
+
+### Exploring Flask-Limiter Functionalities
+
+The Flask-Limiter [documentation](https://flask-limiter.readthedocs.io/en/stable/) showcases a lot of features in the extension. We will briefly illustrate some of them in this section.
+
+#### Setting Default Rate Limit rules
+
+Flask-Limiter provides the functionality to set default rate limit rules that would be automatically applied to every endpoint in your Flask API.
+
+To set default rules, use:
+
+```Python
+limiter = Limiter(
+  app,
+  key_func=get_remote_address,
+  default_limits=["200/day", "50/hour"]
+)
+```
+
+NOTE: If an endpoint defines its rate limit rules when a default already exists, the newly defined rule is applied on the route alone and the defaults are ignored.
```

Comment:
Apply blockquotes here and other parts of the article where applicable.

Reply:
Thank you for highlighting this, just learned about the blockquote feature in MD. I've added it to the article.